src/contracts/core/Slasher.sol

```diff
@@ -1,102 +1,368 @@
+     * @param bipsToIncrease the basis points slashing to modify for given strategies
+     */
+    function increaseRequestedBipsToSlash(
+        address operator,
+        bytes4 operatorSetID,
+        IStrategy[] memory strategies,
+        uint32 bipsToIncrease
+    ) external {
+        require(bipsToIncrease > 0, "Slasher.increaseRequestedBipsToSlash: bipsToIncrease must be positive");
+        require(
+            bipsToIncrease < SlashingAccountingUtils.BIPS_FACTOR,
+            "Slasher.increaseRequestedBipsToSlash: bipsToIncrease must be less than BIPS_FACTOR"
+        );
+        _modifyRequestedBipsToSlash(
+            operator, operatorSetID, strategies, EpochUtils.currentEpoch(), int32(bipsToIncrease)
+        );
+    }
 
-    function optIntoSlashing(address) external {}
+    /**
+     * @notice Called by an AVS to reduce its own slashing request for a given
+     * operator set and operator in the current or previous epoch
+     *
+     * @param operator the operator that the calling AVS is to reduce the bips they want to slash
+     * @param operatorSetID the id of the operator set the AVS is reducing their slashing for
+     * @param strategies the list of strategies slashing requested is being reduced for
+     * @param epoch the epoch in which slashing was requested
+     * @param bipsToReduce the basis points slashing to reduced for given strategies
+     */
+    function reduceRequestedBipsToSlash(
+        address operator,
+        bytes4 operatorSetID,
+        IStrategy[] memory strategies,
+        uint32 epoch,
+        uint32 bipsToReduce
+    ) external {
+        uint32 currentEpoch = EpochUtils.currentEpoch();
+        require(
+            epoch == currentEpoch || epoch + 1 == currentEpoch,
+            "Slasher.reduceRequestedBipsToSlash: can only reduce for current or previous epoch"
+        );
+        require(bipsToReduce > 0, "Slasher.reduceRequestedBipsToSlash: bipsToReduce must be negative");
+        require(
+            bipsToReduce < 2 ** 31, "Slasher.reduceRequestedBipsToSlash: bipsToReduce must be less than minimum int32"
+        );
+        _modifyRequestedBipsToSlash(operator, operatorSetID, strategies, epoch, -int32(bipsToReduce));
+    }
 
-    function freezeOperator(address) external {}
+    /**
+     * @notice Permissionlessly called to execute slashing of a given list of
+     * strategies for a given operator, for the latest unslashed epoch
+     *
+     * @param operator the operator to slash
+     * @param strategies the list of strategies to execute slashing for
+     * @param epoch the epoch in which the slashing requests to execute were made
+     */
+    function executeSlashing(address operator, IStrategy[] memory strategies, uint32 epoch) external {
+        require(
+            EpochUtils.currentEpoch() > EpochUtils.getMinExecutionEpochFromRequestEpoch(epoch),
+            "Slasher.executeSlashing: current epoch must be greater than the minimum execution epoch"
+        );
```

Comment:
strict inequality seems incorrect here? shouldn't equality be OK?

Reply:
Should be >= 
`EpochUtils.currentEpoch() >= EpochUtils.getMinExecutionEpochFromRequestEpoch(epoch)`
the right term returns epoch + 2 so the current epoch has to be at least epoch + 3 which is incorrect.